### PR TITLE
DEPRECATION: Replace or update packages relying on punycode #2862

### PR DESCRIPTION
**Pull request by khushipatil1523**
What kind of change does this PR introduce?

This PR will Replace or update packages relying on punycode #2862

Issue Number:

#2862

Did you update for packages?

Yes

## Summary by CodeRabbit

- **New Features**
	- Added a new dependency for improved linting configuration.
  
- **Updates**
	- Updated multiple development dependencies to enhance project setup and performance.
  
- **Configuration Changes**
	- Modified lint-staged configuration to include TypeScript and JSON files for linting.
	- Introduced an overrides section for package version management.
	- Enhanced ESLint configuration by specifying files to ignore during linting.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "@dicebear/core": "^9.2.2",
     "@emotion/react": "^11.13.3",
     "@emotion/styled": "^11.13.0",
+    "@eslint/config-array": "^0.19.1",
```
[the remark this answers is not in the document]
Comment (khushipatil1523) [reply package.json]: changes done

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by palisadoes** [comment]
Please fix the conflicting file

**Comment by palisadoes** [comment]
You don't have to close the PR when there is a conflicting file.
1. Pull the latest upstream/develop-postgres
2. merge it with your local branch
3. fix the conflicts in your IDE
4. commit 
5. push to the changes to your origin

**Pull request closed without merging**